epochStart/metachain/sovereignValidators_test.go

```diff
@@ -0,0 +1,191 @@
+
+	txHash1 := []byte("txHash1")
+	txHash2 := []byte("txHash2")
+	shardDataCacher := testscommon.NewShardedDataCacheNotifierMock()
+	arguments.DataPool = &dataRetrieverMock.PoolsHolderStub{
+		CurrEpochValidatorInfoCalled: func() dataRetriever.ValidatorInfoCacher {
+			return &vics.ValidatorInfoCacherStub{
+				GetValidatorInfoCalled: func(validatorInfoHash []byte) (*state.ShardValidatorInfo, error) {
+					if bytes.Equal(validatorInfoHash, txHash1) {
+						return svi1, nil
+					}
+					if bytes.Equal(validatorInfoHash, txHash2) {
+						return svi2, nil
+					}
+
+					require.Fail(t, "should not call get for another validator")
+					return nil, nil
+				},
+			}
+		},
+		ValidatorsInfoCalled: func() dataRetriever.ShardedDataCacherNotifier {
+			return shardDataCacher
+		},
+	}
+	vic, _ := NewValidatorInfoCreator(arguments)
+	svic, _ := NewSovereignValidatorInfoCreator(vic)
+
+	body := &block.Body{
+		MiniBlocks: []*block.MiniBlock{
+			{
+				SenderShardID:   core.SovereignChainShardId,
+				ReceiverShardID: core.SovereignChainShardId,
+				Type:            block.PeerBlock,
+				TxHashes: [][]byte{
+					txHash1,
+					txHash2,
+				},
+			},
+			{
+				SenderShardID:   core.SovereignChainShardId,
+				ReceiverShardID: core.SovereignChainShardId,
+				Type:            block.StateBlock,
+				TxHashes: [][]byte{
+					[]byte("txHash3"),
+				},
+			},
+		},
+	}
+	marshalledData := svic.CreateMarshalledData(body)
+	require.Equal(t, 1, len(marshalledData))
+	require.Equal(t, 2, len(marshalledData[common.ValidatorInfoTopic]))
+	require.Equal(t, marshalledSVI1, marshalledData[common.ValidatorInfoTopic][0])
+	require.Equal(t, marshalledSVI2, marshalledData[common.ValidatorInfoTopic][1])
+
+	cachedData1, found := shardDataCacher.SearchFirstData(txHash1)
+	require.True(t, found)
+	require.Equal(t, svi1, cachedData1)
+
+	cachedData2, found := shardDataCacher.SearchFirstData(txHash2)
+	require.True(t, found)
+	require.Equal(t, svi2, cachedData2)
+}
+
+func TestSovereignValidatorInfoCreator_CreateMarshalledDataErrorCases(t *testing.T) {
+
+	t.Run("CreateMarshalledData should return nil body is nil", func(t *testing.T) {
+		t.Parallel()
+
+		arguments := createMockEpochValidatorInfoCreatorsArguments()
+		vic, _ := NewValidatorInfoCreator(arguments)
+		svic, _ := NewSovereignValidatorInfoCreator(vic)
+
+		marshalledData := svic.CreateMarshalledData(nil)
+		require.Nil(t, marshalledData)
+	})
+
+	t.Run("CreateMarshalledData should return empty slice when tx hash does not exist in validator info cacher", func(t *testing.T) {
+		t.Parallel()
+
+		arguments := createMockEpochValidatorInfoCreatorsArguments()
+		localErr := errors.New("local error")
+		arguments.DataPool = &dataRetrieverMock.PoolsHolderStub{
+			CurrEpochValidatorInfoCalled: func() dataRetriever.ValidatorInfoCacher {
+				return &vics.ValidatorInfoCacherStub{
+					GetValidatorInfoCalled: func(validatorInfoHash []byte) (*state.ShardValidatorInfo, error) {
+						return nil, localErr
```

Comment:
Maybe add another `wasCalled = true` here and `require.True` at line 150

Reply:
No need, since I already check the return to be empty, so that wasn't called.

Reply:
This was called every time, that's why marshal was never called and marshalledData is empty. 
I would say makes sense to check this was called at least once, if not for every txHash. But it's not really mandatory.